cf-agent/verify_files.c

```diff
@@ -639,29 +607,23 @@ static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp,
     {
         unsigned char rendered_output_digest[EVP_MAX_MD_SIZE + 1] = { 0 };
         HashString(BufferData(output_buffer), BufferSize(output_buffer), rendered_output_digest, CF_DEFAULT_DIGEST);
+
         if (!HashesMatch(existing_output_digest, rendered_output_digest, CF_DEFAULT_DIGEST))
         {
-            if (a.transaction.action == cfa_warn || DONTDO)
-            {
-                cfPS(ctx, LOG_LEVEL_WARNING, PROMISE_RESULT_WARN, pp, &a,
-                     "Need to update rendering of '%s' from mustache template '%s' but policy is dry-run",
-                     pp->promiser, message);
-                result = PromiseResultUpdate(result, PROMISE_RESULT_WARN);
```

Comment:
This `DONTDO` check and related things have to be preserved.

Reply:
These shouldn't be necessary anymore. With these changes, we're now completely relying on `FinishEditContext` to apply the file edits. And as `FinishEditContext` does already handle "DONTDO" and "cfa_warn", we shouldn't have to replicate these checks.

Reply:
:+1: thanks for the explanation.